src/stdlib_specialfunctions_legendre.f90

```diff
@@ -0,0 +1,72 @@
+submodule (stdlib_specialfunctions) stdlib_specialfunctions_legendre
+    implicit none
+
+contains
+
+    ! derivatives of legegendre polynomials
+    ! unspecified behaviour if n is negative
+    pure elemental module function dlegendre_fp64(n,x) result(dleg)
+        integer, intent(in) :: n
+        real(dp), intent(in) :: x
+        real(dp) :: dleg
+
+        select case(n)
+            case(0)
+                dleg = 0
+            case(1)
+                dleg = 1
+            case default
+                block
+                    real(dp) :: leg_down1, leg_down2, leg
+                    real(dp) :: dleg_down1, dleg_down2
+                    integer :: i 
+
+                    leg_down1  = x
+                    dleg_down1 = 1
+
+                    leg_down2  = 1
+                    dleg_down2 = 0
+
+                    do i = 2, n
+                        leg = (2*i-1)*x*leg_down1/i - (i-1)*leg_down2/i
```

Comment:
I get four warnings on implicit conversion here:
```
../src/stdlib_specialfunctions_legendre.f90:31:31:

   31 |                         leg = (2*i-1)*x*leg_down1/i - (i-1)*leg_down2/i
      |                               1
Warning: Conversion from INTEGER(4) to REAL(8) at (1) [-Wconversion-extra]
../src/stdlib_specialfunctions_legendre.f90:31:51:

   31 |                         leg = (2*i-1)*x*leg_down1/i - (i-1)*leg_down2/i
      |                                                   1
Warning: Conversion from INTEGER(4) to REAL(8) at (1) [-Wconversion-extra]
../src/stdlib_specialfunctions_legendre.f90:31:55:

   31 |                         leg = (2*i-1)*x*leg_down1/i - (i-1)*leg_down2/i
      |                                                       1
Warning: Conversion from INTEGER(4) to REAL(8) at (1) [-Wconversion-extra]
../src/stdlib_specialfunctions_legendre.f90:31:71:

   31 |                         leg = (2*i-1)*x*leg_down1/i - (i-1)*leg_down2/i
      |                                                                       1
Warning: Conversion from INTEGER(4) to REAL(8) at (1) [-Wconversion-extra]
```
Can we make those integer real conversion explicit?